managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ManagedLedgerConfig.java

```diff
@@ -411,7 +411,8 @@ public ManagedLedgerConfig setThrottleMarkDelete(double throttleMarkDelete) {
      *            time unit for retention time
      */
     public ManagedLedgerConfig setRetentionTime(int retentionTime, TimeUnit unit) {
-        this.retentionTimeMs = unit.toMillis(retentionTime);
+        // unit.toMillis(-1) = -60000
+        this.retentionTimeMs = retentionTime != -1 ? unit.toMillis(retentionTime) : -1;
```

Comment:
Maybe we don't need to change this, right now, all negative values are treated as infinite flags.

Reply:
If the result does not match the input, it will be confusing, right?

Reply:
@gaoran10 @BewareMyPower Thanks for this catch. I delete this modification in this PR. 
We can discuss this in the another [PR](https://github.com/apache/pulsar/pull/22023).